cadc-storage-adapter-ad/src/main/java/org/opencadc/inventory/storage/ad/AdStorageAdapter.java

```diff
@@ -0,0 +1,330 @@
+     * @param source The stream from which to read.
+     * @return The storage metadata.
+     * 
+     * @throws IncorrectContentChecksumException If the calculated checksum does not the expected
+     *     checksum as described in newArtifact.
+     * @throws IncorrectContentLengthException If the calculated length does not the expected
+     *     length as described in newArtifact.
+     * @throws ReadException If the client failed to stream.
+     * @throws WriteException If the storage system failed to stream.
+     * @throws StorageEngageException If the adapter failed to interact with storage.
+     * @throws TransientException If an unexpected, temporary exception occurred.
+     */
+    public StorageMetadata put(NewArtifact newArtifact, InputStream source)
+        throws IncorrectContentChecksumException, IncorrectContentLengthException, ReadException,
+            WriteException, StorageEngageException, TransientException {
+        throw new UnsupportedOperationException("not supported");
+    }
+        
+    /**
+     * Delete from storage the artifact identified by storageLocation.
+     * @param storageLocation Identifies the artifact to delete.
+     * 
+     * @throws ResourceNotFoundException If the artifact could not be found.
+     * @throws IOException If an unrecoverable error occurred.
+     * @throws StorageEngageException If the adapter failed to interact with storage.
+     * @throws TransientException If an unexpected, temporary exception occurred. 
+     */
+    public void delete(StorageLocation storageLocation)
+        throws ResourceNotFoundException, IOException, StorageEngageException, TransientException {
+        throw new UnsupportedOperationException("not supported");
+    }
+    
+    /**
+     * Iterator of items ordered by their storageIDs.
+     * @return An iterator over an ordered list of items in storage.
+     * 
+     * @throws StorageEngageException If the adapter failed to interact with storage.
+     * @throws TransientException If an unexpected, temporary exception occurred. 
+     */
+    public Iterator<StorageMetadata> iterator()
+        throws StorageEngageException, TransientException {
+        throw new UnsupportedOperationException("not supported");
+    }
+    
+    /**
+     * Iterator of items ordered by their storageIDs in the given bucket.
+     * @param storageBucket Only iterate over items in this bucket.
+     * @return An iterator over an ordered list of items in this storage bucket.
+     * 
+     * @throws StorageEngageException If the adapter failed to interact with storage.
+     * @throws TransientException If an unexpected, temporary exception occurred. 
+     */
+    public Iterator<StorageMetadata> iterator(String storageBucket)
+        throws StorageEngageException, TransientException {
+        InventoryUtil.assertNotNull(AdStorageQueryUtil.class, "storageBucket", storageBucket);
+        if (storageBucket.length() == 0) {
```

Comment:
In theory, one could pass in an empty string and this condition would pass.  The StringUtil class in `cadc-util` has `hasLength()` and `hasText()` functions that you can pick from.

Reply:
ok